core/src/main/java/org/elasticsearch/index/query/GeoDistanceRangeQueryBuilder.java

```diff
@@ -126,50 +161,234 @@ public GeoDistanceRangeQueryBuilder geoDistance(GeoDistance geoDistance) {
         return this;
     }
 
+    public GeoDistance geoDistance() {
+        return geoDistance;
+    }
+
+    public GeoDistanceRangeQueryBuilder unit(DistanceUnit unit) {
+        this.unit = unit;
+        return this;
+    }
+
+    public DistanceUnit unit() {
+        return unit;
+    }
+
     public GeoDistanceRangeQueryBuilder optimizeBbox(String optimizeBbox) {
         this.optimizeBbox = optimizeBbox;
         return this;
     }
 
+    public String optimizeBbox() {
+        return optimizeBbox;
+    }
+
     public GeoDistanceRangeQueryBuilder coerce(boolean coerce) {
+        if (coerce) {
+            this.ignoreMalformed = true;
+        }
         this.coerce = coerce;
         return this;
     }
 
+    public boolean coerce() {
+        return this.coerce;
+    }
+
     public GeoDistanceRangeQueryBuilder ignoreMalformed(boolean ignoreMalformed) {
-        this.ignoreMalformed = ignoreMalformed;
+        if (coerce == false) {
+            this.ignoreMalformed = ignoreMalformed;
+        }
```

Comment:
I was thinking that it might make sense to do that when all the Geo Queries are refactored so we can ensure its applied correctly across all of them?